relay-config/src/config.rs

```diff
@@ -152,10 +152,10 @@ enum ConfigFormat {
 }
 
 impl ConfigFormat {
-    pub fn extension(&self) -> &'static str {
+    pub fn extensions(&self) -> Vec<&'static str> {
         match self {
-            ConfigFormat::Yaml => "yml",
-            ConfigFormat::Json => "json",
+            ConfigFormat::Yaml => vec!["yaml", "yml"],
+            ConfigFormat::Json => vec!["json"],
```

Comment:
The size of these is small and won't change often, can we use `smallvec` instead? See https://docs.rs/smallvec/latest/smallvec/

Here's an example of its usage in the codebase:
https://github.com/dcariotti/relay/blob/926827a0f42ab9532faf521bb9818284389a3fca/relay-general/src/pii/attachments.rs#L28